fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoDetailActivity.java

```diff
@@ -267,244 +82,49 @@ private void setupBackButton() {
-    setCurrentFragment(
-        getSupportFragmentManager(),
-        R.id.fragment_roll_call,
-        () -> RollCallFragment.newInstance(pk));
-  }
-
-  private void setupElectionFragment() {
-    mViewModel
-        .getOpenElectionFragmentEvent()
-        .observe(
-            this,
-            booleanEvent -> {
-              Boolean event = booleanEvent.getContentIfNotHandled();
-              if (event != null) {
-                Log.d(TAG, "opening new election fragment");
-                setCurrentFragment(
-                    getSupportFragmentManager(),
-                    R.id.fragment_election,
-                    ElectionFragment::newInstance);
-              }
-            });
-  }
-
-  public void setUpWalletMessage() {
-    AlertDialog.Builder builder = new AlertDialog.Builder(this);
+  public static void setUpWalletMessage(Context ctx) {
+    AlertDialog.Builder builder = new AlertDialog.Builder(ctx);
     builder.setTitle("You have to setup up your wallet before connecting.");
     builder.setPositiveButton("Ok", (dialog, which) -> dialog.dismiss());
     builder.show();
   }
 
-  private void setupCastVotesFragment() {
-    mViewModel
-        .getOpenCastVotes()
-        .observe(
-            this,
-            booleanEvent -> {
-              Boolean event = booleanEvent.getContentIfNotHandled();
-              if (event != null) {
-                setCurrentFragment(
-                    getSupportFragmentManager(),
-                    R.id.fragment_cast_vote,
-                    CastVoteFragment::newInstance);
-              }
-            });
-  }
-
-  private void setupElectionResultsFragment() {
-    mViewModel
-        .getOpenElectionResultsEvent()
-        .observe(
-            this,
-            booleanEvent -> {
-              Boolean event = booleanEvent.getContentIfNotHandled();
-              if (event != null) {
-                setCurrentFragment(
-                    getSupportFragmentManager(),
-                    R.id.fragment_election_result,
-                    ElectionResultFragment::newInstance);
-              }
-            });
-  }
-
-  private void setupElectionStartFragment() {
-    mViewModel
-        .getOpenStartElectionEvent()
-        .observe(
-            this,
-            booleanSingleEvent -> {
-              Boolean event = booleanSingleEvent.getContentIfNotHandled();
-              if (event != null) {
-                setCurrentFragment(
-                    getSupportFragmentManager(),
-                    R.id.fragment_election_start,
-                    ElectionStartFragment::newInstance);
-              }
-            });
-  }
-
   public void setupNavigationBar() {
     navbar.setOnItemSelectedListener(
         item -> {
           int id = item.getItemId();
           if (id == R.id.lao_detail_event_list_menu) {
-            setupLaoFragment();
+            setCurrentFragment(
+                getSupportFragmentManager(),
```

Comment:
As `setCurrentFragment` is static, we cannot access members of an object.

And it is static because it is also used outside the class itself.
I figure out a way to do that without providing the `FragmentManager`. If you have any idea, I'd be pleased to hear it.